gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -483,26 +483,33 @@ public static List<Service> parseService(
                 overriddenServiceName =
                     languageSettings.getJavaServiceName(fileDescriptor.getPackage(), s.getName());
               }
-              return serviceBuilder
-                  .setName(serviceName)
-                  .setOverriddenName(overriddenServiceName)
-                  .setDefaultHost(defaultHost)
-                  .setOauthScopes(oauthScopes)
-                  .setPakkage(pakkage)
-                  .setOriginalJavaPackage(originalJavaPackage)
-                  .setProtoPakkage(fileDescriptor.getPackage())
-                  .setIsDeprecated(isDeprecated)
-                  .setMethods(
-                      parseMethods(
-                          s,
-                          pakkage,
-                          messageTypes,
-                          resourceNames,
-                          serviceConfigOpt,
-                          serviceYamlProtoOpt,
-                          outputArgResourceNames,
-                          transport))
-                  .build();
+              Service.Builder service =
+                  serviceBuilder
+                      .setName(serviceName)
+                      .setOverriddenName(overriddenServiceName)
+                      .setDefaultHost(defaultHost)
+                      .setOauthScopes(oauthScopes)
+                      .setPakkage(pakkage)
+                      .setOriginalJavaPackage(originalJavaPackage)
+                      .setProtoPakkage(fileDescriptor.getPackage())
+                      .setIsDeprecated(isDeprecated)
+                      .setMethods(
+                          parseMethods(
+                              s,
+                              pakkage,
+                              messageTypes,
+                              resourceNames,
+                              serviceConfigOpt,
+                              serviceYamlProtoOpt,
+                              outputArgResourceNames,
+                              transport));
+
+              String apiVersion = null;
+              if (serviceOptions.hasExtension(ClientProto.apiVersion)) {
+                apiVersion = serviceOptions.getExtension(ClientProto.apiVersion);
+                service.setApiVersion(apiVersion);
+              }
+              return service.build();
```

Comment:
nit: Since Service's apiVersion is nullable, can we move this extraction section to before the Builder?

```
String apiVersion = null;
if (serviceOptions.hasExtension(ClientProto.apiVersion)) {
  apiVersion = serviceOptions.getExtension(ClientProto.apiVersion);
}

return serviceBuilder
  ...
  .setApiVersion(apiVersion)
  .build()
```